src/main/java/spoon/support/reflect/code/CtTypePatternImpl.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: (MIT OR CECILL-C)
+ *
+ * Copyright (C) 2006-2019 INRIA and contributors
+ *
+ * Spoon is available either under the terms of the MIT License (see LICENSE-MIT.txt) of the Cecill-C License (see LICENSE-CECILL-C.txt). You as the user are entitled to choose the terms under which to adopt Spoon.
+ */
+package spoon.support.reflect.code;
+
+import spoon.reflect.annotations.MetamodelPropertyField;
+import spoon.reflect.code.CtLocalVariable;
+import spoon.reflect.code.CtTypePattern;
+import spoon.reflect.path.CtRole;
+import spoon.reflect.visitor.CtVisitor;
+
+public class CtTypePatternImpl<T> extends CtExpressionImpl<Void> implements CtTypePattern<T> {
+	private static final long serialVersionUID = 1L;
+
+	@MetamodelPropertyField(role = CtRole.VARIABLE)
+	private CtLocalVariable<?> variable;
+
+	@Override
+	public CtLocalVariable<?> getVariable() {
+		return this.variable;
+	}
+
+	@Override
+	public <C extends CtTypePattern<?>> C setVariable(CtLocalVariable<?> variable) {
+		if (variable != null) {
+			variable.setParent(this);
+		}
+		getFactory().getEnvironment().getModelChangeListener()
+				// TODO is this the proper role? New role needed?
+				.onObjectUpdate(this, CtRole.VARIABLE, variable, this.variable);
```

Comment:
This looks like the correct role to me, @monperrus?

Reply:
LGTM